src/lib.rs

```diff
@@ -166,13 +167,30 @@ where
     /// Creates a new rolling file appender with the given condition.
     /// The parent directory of the base path must already exist.
     pub fn new<P>(path: P, condition: RC, max_files: usize) -> io::Result<RollingFileAppender<RC>>
+    where
+        P: AsRef<Path>,
+    {
+        Self::_new(path, condition, max_files, None)
+    }
+
+    /// Creates a new rolling file appender with the given condition and write buffer capacity.
+    /// The parent directory of the base path must already exist.
+    pub fn new_with_buffer_capacity<P>(path: P, condition: RC, max_files: usize, buffer_capacity: usize) -> io::Result<RollingFileAppender<RC>>
```

Comment:
I considered `Appender::with_capacity`, straightly mimicing `BufWriter''s counterpart name. but that would confuse from the view point of `rolling-file` crate users.

also, argument ordering can be contested as well. :)